### PR TITLE
Quickstart optimization

### LINKED ISSUE #720: Quickstart optimization
**What would you like to be added**:

Add a NEW script to start an environment for Quickstart. It contains all infrastructure, control plane, and some samples for test. And organize the flow of installation and integrate scripts.

**Why is this needed**:

Reduce the difficulty of quickstart

### PR DESCRIPTION
**Pull request by lfbear**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Optimize script `hack/local-up-karmada.sh`, add creating member clusters and join to Karmada.
Because original `hack/karmada-bootstrap.sh`  has covered the requirement, changes have done as the following:

- Original `hack/local-up-karmada.sh` --> DEL
- Original `hack/karmada-bootstrap.sh`  --RENAME--> `hack/local-up-karmada.sh`
- Modify the related docs

**Which issue(s) this PR fixes**:
Fixes #720

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by lfbear** [comment]
PTAL @RainbowMango  Thank you

**Comment by RainbowMango** [comment]
Hi Please rebase your code since we have made changes recently.
The functionality of `join` cluster should be moved out from the README too.

**Comment by lfbear** [comment]
> The functionality of `join` cluster should be moved out from the README too.

The part of `join` in README seems to be reserved, shall we add the 'optional' in front? Because the 'remote' install way need it yet

File: samples/nginx/propagationpolicy.yaml [hunk 1]
```diff
@@ -11,3 +11,4 @@ spec:
     clusterAffinity:
       clusterNames:
         - member1
+        - member2
```
Comment (RainbowMango) [inline samples/nginx/propagationpolicy.yaml:14]: This willl change the output above:
```
kubectl get deployment
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   1/1     1            1           43s
```

`1/1` --> `2/1`. That will confuse people that not familiar with Karmada.

Comment (RainbowMango) [reply samples/nginx/propagationpolicy.yaml:14]: My suggestion is to update the PropagationPolicy to `divide` replicas.

File: README.md [hunk 2]
```diff
@@ -108,120 +102,74 @@ There are several demonstrations of common cases.
...
-#### 2. Join member cluster to karmada control plane
-
-You can choose one of mode: [push](#21-push-mode-karmada-controls-the-member-cluster-initiative-by-using-karmadactl) or
-[pull](#22-pull-mode-installing-karmada-agent-in-the-member-cluster), either will help you join a member cluster.
-
-##### 2.1. Push Mode: Karmada controls the member cluster initiative by using `karmadactl`
+#### Network resource requirements:
+Please make sure you can access the following resources, any network problem please refer to the [troubleshooting](/docs/troubleshooting.md#i-cant-access-some-resources-when-install-karmada).
+- Google Container Registry: k8s.gcr.io
+- Download golang package, such as golang.org, k8s.io, etc.
```
Comment (RainbowMango) [inline README.md:148]:
```suggestion
```
It's too late to tell users the network requirements, they might have met the problem before reading these tips.

We might add a summary description at `Prerequisites`. (No further suggestion, for now, we can add it later.)

File: README.md [hunk 3]
```diff
@@ -118,90 +112,40 @@ cd karmada
 
 #### 3. Deploy and run karmada control plane:
 
-run the following script: (It will create a host cluster by kind)
+run the following script:
 
 ```
-hack/local-up-karmada.sh
+# hack/local-up-karmada.sh
```
Comment (RainbowMango) [inline README.md:118]: Still one `#` here.

Comment (lfbear) [reply README.md:118]: this script needs root privilege for installing `kind` and `kubectl`

**Comment by lfbear** [comment]
Re-record the demo which syncs from the latest sample, preview here: https://github.com/lfbear/karmada/tree/pr_quickstart_optimization#demo

**Review by RainbowMango — COMMENTED** [review]
/lgtm
/approve

File: README.md (hunk 3 above)
Comment (RainbowMango) [reply README.md:118]: I don't where requires the `root privilege`, let's figure it out and try to fix it.
I'll open an issue to track it.

Comment (RainbowMango) [reply README.md:118]: Done. https://github.com/karmada-io/karmada/issues/748

**Pull request merged**